### PR TITLE
Fix TSX regression

### PR DESCRIPTION
**Pull request by Nekotekina**

### CODE REVIEW

**Comment by kdlz4161** [comment]
esboco 20 1

**Comment by LuismaSP89** [comment]
@kdlz4161 I think that problem is not related with this PR since it happens in master too, (At least in other games like Resident evil 0 HD, when you enter in the wardrobe of the beginning)

**Comment by spring568** [comment]
I loaded up my save for P5 in front of kamoshida's palace and it freezes on my skylake cpu with the TSX work you've done Nekotekina. This one doesn't seem to fix it either. Hope I'm not just doing something wrong

**Comment by Xcedf** [comment]
For me i found that smth happened between 55b250c1 version which cause games to randomly hang and  2f58c314 last working version of your TSX pr

**Comment by TigiHof** [comment]
Heavy Rain still freezes. 
[RPCS3.log.gz](https://github.com/RPCS3/rpcs3/files/1886356/RPCS3.log.gz)

And I can confirm the statement by Xcedf:  Build 2f58c31 worked much better (less freezes)!

**Comment by maximilian578** [comment]
i5-6500, Vulkan. The hang still persists on P5 and MGS HD Collection.
[RPCS3.log.gz](https://github.com/RPCS3/rpcs3/files/1886418/RPCS3.log.gz)
[RPCS3.log.gz](https://github.com/RPCS3/rpcs3/files/1886420/RPCS3.log.gz)

**Comment by maximilian578** [comment]
i5-6500, The hang still persists on P5 and MGS HD Collection.
[RPCS3.log.gz](https://github.com/RPCS3/rpcs3/files/1886428/RPCS3.log.gz)
[RPCS3.log.gz](https://github.com/RPCS3/rpcs3/files/1886432/RPCS3.log.gz)

**Comment by L1ttl3c0w** [comment]
i7-6700, still experience P5 ingame freezes.

**Comment by miho171** [comment]
i5 7700k with TSX feature, in version 0.0.5-6642 games like P5, Drakengard 3, Skate 3, Lollipop Chainsaw all freeze after 10 min max.

**Comment by L1ttl3c0w** [comment]
Same here, last commit did not fix the freeze issue. I suspect the freeze more likely happens during shader compilation.

**Comment by maximilian578** [comment]
a2f4349 fixed the hang issue i had with P5 and MGS.

**Comment by spyropt** [comment]
Red dead redemption and the last of us just crash the later doesnt even load cpu x5675 without tsx

**Comment by MSuih** [comment]
Am I doing something wrong or is this build causing a ton of regressions? During a quick test Skate 2 & 3 segfault as soon as shader loading is finished, Uncharted 1 shows nothing apart from black screen and Diablo segfaults during main menu. This is on Ryzen 5 1600 & RX 480 .

Here's the error for Skate 2, I can post more if you're interested.
>Unhandled Win32 exception 0xC0000005.
>Segfault reading location 000000040d0075a8 at 0000000000522f71.
>Instruction address: 0000000000522f71.
>Module name: 'neko.exe'.
>Module base: 0000000000010000.
>RPCS3 image base: 0000000000010000.

**Comment by L1ttl3c0w** [comment]
I have TSX (i7-6700) and Diablo crashes here too.

**Comment by L1ttl3c0w** [comment]
Crashes fixed here, P5 freeze still reproducible. Will check again with the current appveyor build in the making.

**Comment by L1ttl3c0w** [comment]
Based on a 10 minutes of playing with P5, the latest commit seems to fix the random freezes.

**Comment by Zangetsu38** [comment]
sport champion now hmm
>F {SPU[0x2000005] Thread (MSTREAMCellSpursKernel0)} MEM: Access violation reading location 0xaaaaaaaa

File: Utilities/sysinfo.h [hunk 1]
```diff
@@ -39,11 +39,11 @@ namespace utils
 
 	bool has_xop();
 
-	inline bool transaction_enter()
+	FORCE_INLINE bool transaction_enter(uint* out = nullptr)
 	{
 		while (true)
 		{
-			const auto status = _xbegin();
+			const uint status = _xbegin();
```
Comment (TGEnigma) [inline Utilities/sysinfo.h]: why uint instead of u32

Comment (Nekotekina) [reply Utilities/sysinfo.h]: It's correct type of the return value, doesn't need casting.

**Comment by spyropt** [comment]
im still seeing regressions with the latest master (this pr)
the last of us does not go into menu/ingame 
red dead redemption hangs on cutscenes
Uncharted drakes fortune hangs right after booting (coin spinning)
skate 3 hangs before going ingame
cpu without tsx/avx (xeon x5675)
i can provide more logs if needed
[RPCS3.log.gz](https://github.com/RPCS3/rpcs3/files/1886696/RPCS3.log.gz)

**Comment by ruthan** [comment]
I tried Skyrim on TSX CPU before intro is shader compilation, there is some endless loop - which using whole CPU.

**Comment by JonySimmer** [comment]
There is a regression for Until Dawn. The game freezes before the main menu.

**Comment by Zangetsu38** [comment]
No fixed at all
Sport champion is more regressed
>F {SPU[0x2000005] Thread (MSTREAMCellSpursKernel0)} class std::runtime_error thrown: Invalid thread type (cmd=0x41, lsa=0x2740f, ea=0xffffffff, tag=0x1, size=0x7def)
(in file h:\git\rpcs3-dante\rpcs3\emu\cell\sputhread.cpp:604)

**Comment by Rpcs3tester** [comment]
its still hang with this PR

Siren

heavenly sword 

( i got more speed with Tekken Tag Tournament 2 )

**Comment by spring568** [comment]
@Nekotekina Thanks for the fix, P5 is working! :3

**Comment by miho171** [comment]
emu crashing when trying to open trophies menu
rpcs3err

**Comment by NPJB00512** [comment]
Gundam EXVSFB can't play and Audio 

F {PPU[0x100005d] Thread
(CnuSound2Thread:243ffd90) [0x0045ac68]} MEM: Access violation writing location 0x0
[RPCS3.log.gz](https://github.com/RPCS3/rpcs3/files/1886791/RPCS3.log.gz)

**Comment by al0xf** [comment]
For regressions open a new issue and prove where it happened, that is log file from the last working build, and log file from the first broken build. 

Games that never actually worked do not matter. If TLoU crashes in 3 seconds or 30 seconds is irrelevant, it's all the same race conditions.

**Pull request merged**